archaius2-archaius1-bridge/src/main/java/com/netflix/archaius/bridge/AbstractConfigurationBridge.java

```diff
@@ -0,0 +1,133 @@
+package com.netflix.archaius.bridge;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import org.apache.commons.configuration.AbstractConfiguration;
+import org.apache.commons.configuration.Configuration;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+import com.netflix.archaius.Config;
+import com.netflix.archaius.commons.CommonsToConfig;
+import com.netflix.archaius.config.CompositeConfig;
+import com.netflix.archaius.config.SettableConfig;
+import com.netflix.archaius.exceptions.ConfigException;
+import com.netflix.archaius.inject.LibrariesLayer;
+import com.netflix.archaius.inject.RuntimeLayer;
+import com.netflix.config.AggregatedConfiguration;
+import com.netflix.config.DeploymentContext;
+
+/**
+ * @see StaticArchaiusBridgeModule
+ * @author elandau
+ */
+@Singleton
+class AbstractConfigurationBridge extends AbstractConfiguration implements AggregatedConfiguration {
+
+    private final Config config;
+    private final SettableConfig settable;
+    private final CompositeConfig libraries;
+    private final AtomicInteger libNameCounter = new AtomicInteger();
+    
+    @Inject
+    public AbstractConfigurationBridge(
+            Config config, 
+            @LibrariesLayer CompositeConfig libraries, 
+            @RuntimeLayer SettableConfig settable, 
+            DeploymentContext context) {
+        this.config = config;
+        this.settable = settable;
+        this.libraries = libraries;
+    }
+    
+    @Override
+    public boolean isEmpty() {
+        return config.isEmpty();
+    }
+
+    @Override
+    public boolean containsKey(String key) {
+        return config.containsKey(key);
+    }
+
+    @Override
+    public Object getProperty(String key) {
+        return config.getRawProperty(key);  // Should interpolate
+    }
+
+    @Override
+    public Iterator<String> getKeys() {
+        return config.getKeys();
+    }
+
+    @Override
+    protected void addPropertyDirect(String key, Object value) {
+        settable.setProperty(key, value);
+    }
+
+    @Override
+    public void addConfiguration(AbstractConfiguration config) {
+        addConfiguration(config, "Config-" + libNameCounter.incrementAndGet());
+    }
+
+    @Override
+    public void addConfiguration(AbstractConfiguration config, String name) {
+        try {
+            libraries.addConfig(name, new CommonsToConfig(config));
```

Comment:
Specifying the order will be a setting for the @LibrariesLayer CompositeConfig that is done in the archaius-guice integration and will therefore be separate from the bridge.  It will be an application decision.  So, for internal Netflix we would set this up with reversed order to match the behavior of NetflixConfiguration.  This will be the case both for the bridge as well as regular use of archaius2.